t/t5555-http-smart-common.sh

```diff
@@ -123,9 +123,21 @@ test_expect_success 'git receive-pack --advertise-refs: v1' '
 '
 
 test_expect_success 'git upload-pack --advertise-refs: v2' '
+	printf "agent=FAKE" >agent_os_name &&
```

Comment:
Actually I think "agent_and_os_version" would be a better filename than "agent_os_name". (This is true everywhere below.)